fireant/queries/builder.py

```diff
@@ -328,17 +405,22 @@ def fetch(self, hint=None, force_include=()) -> pd.Series:
         query = add_hints(self.sql, hint)[0]
 
         dimension = self._dimensions[0]
-        definition = dimension.definition.as_(alias_selector(dimension.alias))
+        alias_definition = dimension.definition.as_(alias_selector(dimension.alias))
+        dimension_definition = dimension.definition
+
+        if self.dataset.hint_table:
+            alias_definition = alias_definition.replace_table(self.dataset.table, self.dataset.hint_table)
+            dimension_definition = dimension.definition.replace_table(self.dataset.table, self.dataset.hint_table)
```

Comment:
i think you might still need to do `replace_table` at the query level so that all of the filters and joins etc also have their columns replaced, not sure